clang/lib/AST/ByteCode/Compiler.cpp

```diff
@@ -4674,10 +4677,7 @@ VarCreationState Compiler<Emitter>::visitVarDecl(const VarDecl *VD,
       if (!visitInitializer(Init))
         return false;
 
-      if (!this->emitFinishInit(Init))
-        return false;
-
-      return this->emitPopPtr(Init);
+      return this->emitFinishInitGlobal(Init);
```

Comment:
Is this change related? Can you explain what's going on a bit more?

Reply:
Yes, `FinishInitGlobal` will copy the value to new memory that's been allocated in the `Program` and won't be freed at the end of the evaluation.